lib/iris/tests/unit/fileformats/pp/test_PPField.py

```diff
@@ -102,6 +113,45 @@ def field_checksum(data):
             'If float64 precision is required then please save in a '
             'different format')
 
+    def test_mdi_value_warning(self):
+        # Check that an unmasked MDI value raises a warning.
+        field = TestPPField()._ready_for_save()
+        field.bmdi = -9.87e12
+        # Make float32 data, as float64 default produces an extra warning.
+        field.data = np.ma.masked_array([1., field.bmdi, 3.], dtype=np.float32)
+        mock_warn = self.patch('warnings.warn')
+        with self.temp_filename('.pp') as temp_filename:
+            with open(temp_filename, 'wb') as pp_file:
+                field.save(pp_file)
+        mock_warn.assert_called_once_with(
+            'Masked PPField data contains unmasked points equal to '
+            'the fill value, -9.87e+12.')
+
+    def test_mdi_masked_value_nowarning(self):
+        # Check that a *masked* MDI value does not raise a warning.
+        field = TestPPField()._ready_for_save()
+        field.bmdi = -9.87e12
+        # Make float32 data, as float64 default produces an extra warning.
+        field.data = np.ma.masked_array([1., field.bmdi, 3.], dtype=np.float32)
+        field.data[1] = np.ma.masked
+        mock_warn = self.patch('warnings.warn')
+        with self.temp_filename('.pp') as temp_filename:
+            with open(temp_filename, 'wb') as pp_file:
+                field.save(pp_file)
+        mock_warn.assert_not_called()
+
+    def test_plainarray_mdi_value_nowarning(self):
+        # Check that MDI in *unmasked* data does not raise a warning.
```

Comment:
If we have a point==mdi in *un*-masked data, I'd assume that means the data was loaded by non-standard means, and the magic value means it really *is* a missing point.